landlab/components/river_bed_dynamics/river_bed_dynamics.py

```diff
@@ -0,0 +1,3433 @@
+
+        # Copies the grain sizes after removing sand if Parker Eq. is used.
+        # All grains will be copies if other equation is used.
+        self._grain_size_D_original = grain_size_D
+
+        # Grain sizes frequency - Now is not cumulative anymore and has the
+        # same dimensions as the equivalent grain_size_D or grain_size_D_equivalent
+        grain_size_frequency = np.abs(-np.diff(grain_size_frequency / 100, axis=0))
+        # Equivalent grain sizes
+        grain_size_D_equivalent = (grain_size_D[0:-1] * grain_size_D[1:]) ** 0.5
+        sand_fraction = np.zeros_like(
+            self._bed_surface__grain_size_distribution_location_node
+        )  # Sand fraction at each node
+
+        # Bed grain sizes frequency in each node
+        grain_size_D_equivalent_frequency = np.zeros(
+            [self._grid.number_of_nodes, grain_size_D.shape[0] - 1]
+        )
+        for i in range(number_gsd_locations):
+            (id_gsdLocation,) = np.where(
+                self._bed_surface__grain_size_distribution_location_node == i
+            )
+            sand_fraction[id_gsdLocation] = sand_fraction_0[i] / 100
+            grain_size_D_equivalent_frequency[id_gsdLocation, :] = grain_size_frequency[
+                :, i
+            ]
+
+        # Calculating D50 at each node - Grain sizes in Psi scale
+        grain_size_Psi_scale_D = np.flip(np.log2(grain_size_D), axis=0)
+        # Frequency of each grain size in each node
+        grain_size_D_equivalent_frequency_cumulative = np.hstack(
+            (
+                np.zeros(
+                    [
+                        self._bed_surface__grain_size_distribution_location_node.shape[
+                            0
+                        ],
+                        1,
+                    ]
+                ),
+                np.cumsum(np.flip(grain_size_D_equivalent_frequency, axis=1), axis=1),
+            )
+        )
+
+        # Finds the index of the grain size smaller than 50% in each node
+        i0 = np.argmin(grain_size_D_equivalent_frequency_cumulative <= 0.5, axis=1) - 1
+        # Finds the index of the grain size larger than 50% in each node
+        i1 = np.argmax(grain_size_D_equivalent_frequency_cumulative > 0.5, axis=1)
+        nodes_list = np.arange(0, self._grid.number_of_nodes)
+
+        grain_size_Psi_scale_D50 = grain_size_Psi_scale_D[i0] + (
+            (grain_size_Psi_scale_D[i1] - grain_size_Psi_scale_D[i0])
+            / (
+                grain_size_D_equivalent_frequency_cumulative[nodes_list, i1]
+                - grain_size_D_equivalent_frequency_cumulative[nodes_list, i0]
+            )
+        ) * (0.5 - grain_size_D_equivalent_frequency_cumulative[nodes_list, i0])
+        D50 = 2**grain_size_Psi_scale_D50  # Median grain size in each node
+
+        # Calculating the geometric mean and standard deviation
+        # Equivalent grain sizes in Psi scale
+        Psi_grain_size_D_equivalent = np.log2(grain_size_D_equivalent)
```

Comment:
Variable names shouldn't start with a capital letter—notice that *GitHub* thinks it's a class and so highlights it as such.

Reply:
Thanks, done!